src/views/Nft/market/hooks/useCollectionNfts.tsx

```diff
@@ -0,0 +1,163 @@
+import { useEffect, useState } from 'react'
+import { ApiResponseCollectionTokens, NftAttribute, NftToken } from 'state/nftMarket/types'
+import { useGetNftFilters, useGetNftOrdering, useGetNftShowOnlyOnSale } from 'state/nftMarket/hooks'
+import {
+  fetchNftsFiltered,
+  getMarketDataForTokenIds,
+  getNftApi,
+  getNftsFromCollectionApi,
+  getNftsMarketData,
+} from 'state/nftMarket/helpers'
+import isEmpty from 'lodash/isEmpty'
+import uniqBy from 'lodash/uniqBy'
+import { REQUEST_SIZE } from '../Collection/config'
+
+export const useCollectionNfts = (collectionAddress: string) => {
+  const [isFetchingNfts, setIsFetchingNfts] = useState(false)
+  const [nfts, setNfts] = useState<NftToken[]>([])
+  const [page, setPage] = useState(1)
+  const { field, direction } = useGetNftOrdering(collectionAddress)
+  const showOnlyNftsOnSale = useGetNftShowOnlyOnSale(collectionAddress)
+  const nftFilters = useGetNftFilters(collectionAddress)
+  const [itemListingSettings, setItemListingSettings] = useState({
+    page: 1,
+    field,
+    direction,
+    showOnlyNftsOnSale,
+    nftFilters,
+  })
+
+  const itemListingSettingsJson = JSON.stringify(itemListingSettings)
+  const filtersJson = JSON.stringify(nftFilters)
```

Comment:
is this stringify and parse to avoid useEffect necessary? I thought it won't re-run if the state is not changed

Reply:
Filters can trigger infinite rerender if there is no data stored in redux see other pr for that (https://github.com/pancakeswap/pancake-frontend/pull/3119)

Stringfy Item listings to avoid trigger fetching useEffect multiple times on mount, since all other effects triggered and setting a new object on mount causes useEffect that fetching nfts triggered multiple times.